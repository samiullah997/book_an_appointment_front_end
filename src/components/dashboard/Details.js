import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { Link, useNavigate } from 'react-router-dom';
import { singleBike } from '../../redux/reducer/bikeReducer';

const Details = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const userData = localStorage.getItem('bookBikeUser');
  const user = JSON.parse(userData);
  if (!user) {
    navigate('/');
  }

  const bikeData = useSelector((state) => state.bikeReducer);
  const { bike } = bikeData;
  return (
    <div className="container flex flex-row w-full m-28">
      <div className="container flex justify-center py-2  items-center">

        <img src={bike.bike_image} className="w-full h-full" alt="details" />

      </div>
      <div className="conatiner w-full h-auto flex flex-col items-end py-4  px-30">
        <h1 className="text-2xl font-bold py-4">{bike.name}</h1>
        <div className="flex flex-row justify-between w-60 border bg-gray-100 px-2 py-2">
          <span>Price</span>
          <span>
            $

            {bike.daily_rate}
          </span>
        </div>
        <h1 className="text-2xl font-bold py-2  px-4">{bike.brand}</h1>
        <p className="text-2xl font-bold py-2  px-4">{bike.bike_type}</p>
        <Link to="/user/Reservation" onClick={() => dispatch(singleBike(bike.id))}>
          <button type="button" className="bg-green-500 mt-40 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full">Reserve</button>

        </Link>
      </div>
    </div>
  );
};
export default Details;
