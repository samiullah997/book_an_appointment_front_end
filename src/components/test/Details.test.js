import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/Store';
import Details from '../dashboard/Details';

describe('Details', () => {
  it('renders correctly', () => {
    const tree = render(<Provider store={store}><Router><Details /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
