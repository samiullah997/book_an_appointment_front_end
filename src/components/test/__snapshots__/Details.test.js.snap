// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container flex flex-row w-full m-28"
      >
        <div
          class="container flex justify-center py-2  items-center"
        >
          <img
            alt="details"
            class="w-full h-full"
          />
        </div>
        <div
          class="conatiner w-full h-auto flex flex-col items-end py-4  px-30"
        >
          <h1
            class="text-2xl font-bold py-4"
          />
          <div
            class="flex flex-row justify-between w-60 border bg-gray-100 px-2 py-2"
          >
            <span>
              Price
            </span>
            <span>
              $
            </span>
          </div>
          <h1
            class="text-2xl font-bold py-2  px-4"
          />
          <p
            class="text-2xl font-bold py-2  px-4"
          />
          <a
            href="/user/Reservation"
          >
            <button
              class="bg-green-500 mt-40 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full"
              type="button"
            >
              Reserve
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container flex flex-row w-full m-28"
    >
      <div
        class="container flex justify-center py-2  items-center"
      >
        <img
          alt="details"
          class="w-full h-full"
        />
      </div>
      <div
        class="conatiner w-full h-auto flex flex-col items-end py-4  px-30"
      >
        <h1
          class="text-2xl font-bold py-4"
        />
        <div
          class="flex flex-row justify-between w-60 border bg-gray-100 px-2 py-2"
        >
          <span>
            Price
          </span>
          <span>
            $
          </span>
        </div>
        <h1
          class="text-2xl font-bold py-2  px-4"
        />
        <p
          class="text-2xl font-bold py-2  px-4"
        />
        <a
          href="/user/Reservation"
        >
          <button
            class="bg-green-500 mt-40 hover:bg-green-700 text-white font-bold py-2 px-4 rounded-full"
            type="button"
          >
            Reserve
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
